신고하기 modal

olivyou2 opened the pull request:
- create Alert.js
- ... and accept into List.js (Report)

![dialog](https://user-images.githubusercontent.com/36305517/215540432-bbaafa85-0dc9-4c4c-bfd4-0172269a80b6.PNG)

olivyou2 committed:
feat: create alert.js and accept to List.js

yuraup commented on src/components/Alert.js:
```diff
@@ -0,0 +1,135 @@
+  height: 100%;
+
+  background: rgba(0, 0, 0, 0.2);
+`;
```
여기 한 칸 띄워주세요

olivyou2 committed:
style: alert.js, modal.js

syeon2565 commented on src/components/Alert.js:
```diff
@@ -0,0 +1,136 @@
+  const outside = useRef();
+  const inside = useRef();
+
+  const clickYes = () => {
+    if (onYes) {
+      onYes();
+    }
+
+    modalClose();
+  };
+
+  const clickNo = () => {
+    if (onNo) {
+      onNo();
+    }
+
+    modalClose();
+  };
```
조금 더 의미있는 함수명 작성해주시면 좋을 것 같습니다!

syeon2565 commented on src/components/Alert.js:
```diff
@@ -0,0 +1,136 @@
+  justify-content: space-between;
+`;
+
+const YesBtn = styled.div`
+  width: 147px;
+  height: 54px;
+  background-color: white;
+  border-radius: 10px;
+
+  display: flex;
+  justify-content: center;
+  align-items: center;
+
+  font-size: 22px;
+`;
+
+const NoBtn = styled.div`
+  width: 147px;
+  height: 54px;
+  background-color: white;
+  border-radius: 10px;
+
+  display: flex;
+  justify-content: center;
+  align-items: center;
+
+  font-size: 22px;
+`;
+
```
버튼인데 div라서 버튼으로 고쳐주시면 좋을 것 같아요!

syeon2565 commented on src/components/List.js:
```diff
@@ -124,7 +127,9 @@ const List = () => {
                 <ReportBox
                   onClick={() => {
-                    reportPrompt(item._id);
+                    //reportPrompt(item._id);
```
주석 사용하지 않는다면 지워주세요!

syeon2565 commented on src/components/List.js:
```diff
@@ -137,6 +142,16 @@ const List = () => {
+            <Alert
+              title={'신고하기'}
+              dialog={alertModalDialog}
+              onYes={async () => {
+                await report.mutateAsync(reportItemId);
+              }}
```
온클릭 함수는 따로 함수를 만들어서 함수만 호출해주세요!

olivyou2 committed:
style: alert.js modal btn style

olivyou2 committed:
refactor: alert.js parameter, variable name change

olivyou2 committed:
refactor: list.js

- remove unmeaningful comment
- change onYes handler of Alert into external function

olivyou2 committed:
chore: merge into develop
